slidetoact/src/main/java/com/ncorti/slidetoact/SlideToActView.kt

```diff
@@ -735,6 +743,31 @@ class SlideToActView @JvmOverloads constructor(
         animSet.start()
     }
 
+    /** Private method to update view to complete state immediately after view created */
+    private fun setCompleteState() {
+        mPosition = mAreaWidth - mAreaHeight
+        mActualAreaMargin = mAreaHeight / 2
+        mActualAreaWidth = mPosition / 2
+        mIsCompleted = true
+        isEnabled = false
+        val tickListener = ValueAnimator.AnimatorUpdateListener {
+            // We need to enable the drawing of the AnimatedVectorDrawable before starting it.
+            if (!mFlagDrawTick) {
+                mFlagDrawTick = true
+                mTickMargin = mIconMargin
+            }
+        }
+        val tickAnimator: ValueAnimator = createIconAnimator(this, mDrawableTick, tickListener)
+        tickAnimator.start()
+    }
+
+    /**
+     * Method for complete slider immediately without animation
+     */
+    fun setCompleted() {
+        this.mIsCompleted = true
+    }
+
     /**
      * Method that completes the slider
```

Comment:
We do have a API problem here as we have both:

```
fun setCompleted(true)
fun completeSlider()
fun setCompleted(false)
fun resetSlider()
```

Ideally we have something like:

```
fun setCompleted(completed: Boolean, withAnimation: Boolean)

@Deprecate fun completeSlider()
@Deprecate fun resetSlider()
```

Reply:
`fun setCompleted(completed: Boolean, withAnimation: Boolean) `was added